File: scratch_fix_literals.js
const fs = require('fs');
const path = require('path');

const reposDir = 'c:/innova/ecommerce/app/Repositories/Admin';
const repos = fs.readdirSync(reposDir).filter(f => f.endsWith('Repository.php'));

for (const r of repos) {
    const rpath = path.join(reposDir, r);
    let content = fs.readFileSync(rpath, 'utf8');
    const original = content;

    content = content.replace(/public function create\(array \$1 = \[\]\): mixed/g, 'public function create(array $data = []): mixed');
    content = content.replace(/public function update\(\$1, array \$2 = \[\]\): mixed/g, 'public function update($id, array $data = []): mixed');
    content = content.replace(/public function update\(\$2, array \$1 = \[\]\): mixed/g, 'public function update($id, array $data = []): mixed');

    // Also to be safe, there might be places that use `$user` instead of `$data`? No, since I replaced blindly, let's just make it `$data` since that's what was inside.
    // wait, what about the function body? The function body used the original variable name!
    // Since I only replaced the definition, the body still has whatever it had (like `$data`).
    // So `$data` is mostly correct! Let's check `AdminRepository.php` to see if `$data` is used in the body.
    if (content !== original) {
        fs.writeFileSync(rpath, content, 'utf8');
    }
}
console.log('Fixed literal variables!');

File: scratch_fix_literals.py
import os
import re

repos_dir = 'c:/innova/ecommerce/app/Repositories/Admin'
repos = [f for f in os.listdir(repos_dir) if f.endswith('.php') and f.endswith('Repository.php')]

for r in repos:
    rpath = os.path.join(repos_dir, r)
    with open(rpath, 'r', encoding='utf-8') as f:
        content = f.read()

    original = content
    
    # create
    content = content.replace('public function create(array $1 = []): mixed', 'public function create(array $data = []): mixed')

    # update
    content = content.replace('public function update($1, array $2 = []): mixed', 'public function update($id, array $data = []): mixed')
    content = content.replace('public function update($2, array $1 = []): mixed', 'public function update($id, array $data = []): mixed')

    if content != original:
        with open(rpath, 'w', encoding='utf-8') as f:
            f.write(content)
print('Done!')
